cm2350/e200_intc.py

```diff
@@ -153,8 +150,15 @@ def handleException(self):
 
                 # Check if there are any peripheral-specific callbacks for this
                 # exception type
-                for callback in self._callbacks.get(newexc.source, []):
-                    callback(newexc)
+                ######################T SHIS IS SOME JACKED UP CRAP....  .source doesn't exist on more exceptions!
+                if newexc.__ivor__ == EXC_EXTERNAL_INPUT:
+                    for callback in self._callbacks_ext.get(newexc.source, []):
+                        callback(newexc)
+
+                else: 
+                    if newexc.__ivor__ in self._callbacks:
+                        for callback in self._callbacks.get(newexc.__ivor__, []):
+                            callback(newexc)
```

Comment:
This has been addressed in PR #1, callbacks added to external exceptions now use the exception type instead of trying to use the source (which doesn't exist in the MCU exceptions).